pinot-core/src/main/java/com/linkedin/pinot/core/io/reader/impl/v1/VarByteSingleValueReader.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.io.reader.impl.v1;
+
+import com.linkedin.pinot.core.io.compression.ChunkUncompressor;
+import com.linkedin.pinot.core.io.reader.BaseSingleColumnSingleValueReader;
+import com.linkedin.pinot.core.io.reader.impl.VarByteReaderContext;
+import com.linkedin.pinot.core.io.writer.SingleColumnSingleValueWriter;
+import com.linkedin.pinot.core.segment.memory.PinotDataBuffer;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+
+
+/**
+ * Reader class for data written out by {@link SingleColumnSingleValueWriter}.
+ */
+public class VarByteSingleValueReader extends BaseSingleColumnSingleValueReader<VarByteReaderContext> {
+  private static final int INT_SIZE = Integer.SIZE / Byte.SIZE;
+  private static final Charset UTF_8 = Charset.forName("UTF-8");
+
+  final PinotDataBuffer _dataBuffer;
+  final PinotDataBuffer _header;
+  private final ChunkUncompressor _chunkUncompressor;
+  private final int _maxChunkSize;
+
+  private final int _numDocsPerChunk;
+  private final int _numChunks;
+  private final int _lengthOfLongestEntry;
+
+  // Thread local (reusable) byte[] to read bytes from data file.
+  private final ThreadLocal<byte[]> _reusableBytes = new ThreadLocal<byte[]>() {
+    @Override
+    protected byte[] initialValue() {
+      return null;
```

Comment:
Initially I was thinking of having this as static, which doesn't work obviously. Once I removed static, I missed to remove the lazy allocation, fixed now.